NativePlugin/code/build.gradle

```diff
@@ -0,0 +1,50 @@
+apply plugin: 'com.android.library'
+
+android {
+  compileSdkVersion 25
+  buildToolsVersion '25.0.2'
+
+  defaultConfig {
+    externalNativeBuild {
+      cmake {
+        targets 'Earcut'
+        cppFlags '-std=c++11'
+        arguments '-DANDROID_TOOLCHAIN=clang',
+                  '-DANDROID_STL=c++_shared'
+                    cppFlags '-std=c++14',
+                  '-pedantic',
+                  '-fPIC'
+      }
+    }
+  }
+
+  externalNativeBuild {
+    cmake {
+      path '../CMakeLists.txt'
+    }
+  }
+
+  buildTypes {
+    release {
+      externalNativeBuild {
+        cmake.cppFlags'-g0'
+      }
+    }
+  }
+
+  productFlavors {
+    full {
+      externalNativeBuild.cmake.abiFilters 'armeabi-v7a', 'x86'
+    }
+  }
+}
+
+task installLibraries(type: Copy)
+
+installLibraries {
+  description = 'Install libraries into the Unity plugin project...'
+  from 'build/outputs/aar/'
+  into '../../Assets/Plugins/Android/'
+  include 'code-full-release.aar'
+  rename 'code-full-release.aar', 'Earcut.aar'
```

Comment:
I tried this same change on Friday! It definitely cleans up a lot of files. However I ran into trouble making an Android build with the AAR at first. I have two JVMs on my machine: the shell that I used to build the AAR uses Java 8 and Unity defaulted to Java 7. This made Android builds fail in confusing ways.

I'd like to try some scenarios with this AAR and make sure users won't run into similar problems. I suspect that an AAR built with Java 7 may be compatible with a Java 8 build tool, but not the other way around. Can you tell me what version(s) of Java are on your machine @karimnaaji?